src/parser.ts

```diff
@@ -53,7 +53,21 @@ function parseThumbnailToImageItem(data: Thumbnail[], alt: string): ImageItem |
 }
 
 function parseEmojiToImageItem(data: MessageEmoji): ImageItem | undefined {
-  return parseThumbnailToImageItem(data.emoji.image.thumbnails, data.emoji.shortcuts.shift()!)
+  if (data.hasOwnProperty('emoji')) {
+    if (data.emoji.hasOwnProperty('image')) {
+      if (data.emoji.image.hasOwnProperty('thumbnails')) {
+        if (data.emoji.hasOwnProperty('shortcuts')) {
+          if (data.emoji.shortcuts.length > 1) {
+            const emoChar: any = data.emoji.shortcuts.shift();
+            return parseThumbnailToImageItem(data.emoji.image.thumbnails, emoChar);
+          }
+        }
+      }
+    }
+  }
+  else {
+    return;
+  }
```

Comment:
- Use [Optional Chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining) from TypeScript
- Avoid unnecessary nesting
- No need `else` if `undefined` to be returned (no `return` called = `undefined` returned)
```suggestion
  const thumbnailsValid = Array.isArray(data?.emoji?.image?.thumbnails)
  const shortcutsPresent = Array.isArray(data?.emoji?.shortcuts) && data.emoji.shortcuts.length > 0
  if (thumbnailsValid && shortcutsPresent) {
    return parseThumbnailToImageItem(data.emoji.image.thumbnails, data.emoji.shortcuts.shift()!)
  }
```

Reply:
@PikachuEXE you don't need optional chaining between `data` and `emoji`, as data should always have a value.

Reply:
Should probably make this change yourself

Reply:
Sorry - I've been away - is that to me?

Reply:
> Sorry - I've been away - is that to me?

Ahh, thought you might have went inactive. If you are able to make the change then that'd be great! :)